.github/workflows/autorelease.sh

```diff
@@ -0,0 +1,68 @@
+if [ $# -lt 1 ]; then
+    echo -e "Missing parameter. Usage: ./autorelease [github_token]"
+    exit 1
+fi
+
+CHANGELOG_FILE="CHANGELOG.md"
+
+git fetch --prune --unshallow --tags
+
+# Sort tags by date and pick the last one
+LAST_TAG_RAW=$(git for-each-ref --sort=creatordate --format '%(refname:short)' refs/tags | tail -1)
+
+# Remove 'v' in front of the tag
+LAST_TAG="${LAST_TAG_RAW:1}"
+
+# Find the tag's corresponding line in the CHANGELOG
+LINE_BEGIN=$(grep -n "## \[${LAST_TAG}\]" ${CHANGELOG_FILE} | cut -d : -f 1)
+
+# Start at the first paragraph instead of the title
+LINE_BEGIN=$((${LINE_BEGIN} + 2))
+
+# If there is un "Unreleased" title, the 2nd most recent tag is the 3rd title
+grep "## [Unreleased]" ${CHANGELOG_FILE}
+TAG_NUMBER=2
+if [ $? -eq 0 ]; then
+    TAG_NUMBER=3
+fi
+
+CHANGELOG_TITLES=$(grep -n "## \[" ${CHANGELOG_FILE})
+LINE_END=$(echo "${CHANGELOG_TITLES}" | cut -d $'\n' -f ${TAG_NUMBER} | cut -d : -f 1)
+
+if [ -z ${LINE_END} ]; then
+    # Only one title, so the changelog goes until the end of the file
+    LINE_END=$(($(cat ${CHANGELOG_FILE} | wc -l) + 1))
```

Comment:
not really because there are links at the bottom of the file!

I think you can look for the next \[.*\] pattern in the file after your tag. It will match either the previous tag or a link at the bottom...